### PR TITLE
Reconstruct function definition for magic variables in overload error message.

### PR DESCRIPTION
**Pull request by chriseth**
Follow-up #7705

### CODE REVIEW

File: libsolidity/analysis/TypeChecker.cpp [hunk 1]
```diff
@@ -2359,7 +2359,18 @@ bool TypeChecker::visit(Identifier const& _identifier)
...
 
 				for (Declaration const* declaration: annotation.overloadedDeclarations)
-					ssl.append("Candidate:", declaration->location());
+					if (declaration->location().isEmpty())
+					{
+						// Try to re-construct function definition
+						string description;
+						for (auto const& param: declaration->functionType(true)->parameterTypes())
+							description += (description.empty() ? "" : ", ") + param->toString(false);
+						description = "function " + _identifier.name() + "(" + description + ")";
```
Comment (ekpyron) [inline libsolidity/analysis/TypeChecker.cpp:2368]: Do we need ``function ``? Or do we want to use ``builtin ``? Hm - ``function `` is fine I guess.

**Review by ekpyron — DISMISSED** [review]
Makes sense.

**Comment by ekpyron** [comment]
The test failures are weird... is that connected to some recent issue or PR I saw about semver?

**Comment by chriseth** [comment]
Fixed (I hope).

**Review by ekpyron — APPROVED** [review]
Reapproved.

**Pull request merged**